scan.sh

```diff
@@ -1,35 +1,62 @@
 #!/bin/bash
-set -eux
+set -euo pipefail
 
 MICROSCANNER_TOKEN="${MICROSCANNER_TOKEN:-}"
 DOCKER_IMAGE="${1:-}"
-TEMP_IMAGE_TAG=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1 | tr '[:upper:]' '[:lower:]')
-set -o pipefail
+TEMP_IMAGE_TAG=$(tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 32 | head -n 1 | tr '[:upper:]' '[:lower:]' || true)
 
 main() {
+  local MICROSCANNER_BINARY MICROSCANNER_SOURCE
   [[ -z ${MICROSCANNER_TOKEN} ]] && { print_usage; exit 1; }
   [[ -z ${DOCKER_IMAGE} ]] && { print_usage; exit 1; }
 
-  DOCKERFILE=$(mktemp -p . Dockerfile.XXXXXXXXXX)
-  trap cleanup_dockerfile EXIT
+  trap cleanup EXIT
 
-  cat >${DOCKERFILE} <<EOL
+  TEMP_DIR=$(mktemp -d)
+  cd "${TEMP_DIR}"
+
+  MICROSCANNER_SOURCE="https://get.aquasec.com/microscanner"
+  if MICROSCANNER_BINARY=$(command -v microscanner); then
```

Comment:
is those only to allow for an offline build?

Reply:
It saves downloading the image every time if running in a loop, might be worth a flag or documentation note now you mention it...

Reply:
i think the silent nature of this might mean it's unexpected to people. would prefer an cmdline arg or envvar USE_LOCAL_MICROSCANNER or some such. but let's not let the best be the enemy of the good!